lib/private/AppConfig.php

```diff
@@ -30,413 +33,1378 @@
+			};
+		} catch (UnhandledMatchError) {
+			throw new AppConfigIncorrectTypeException('Unknown type ' . $type);
+		}
+	}
+
+	/**
+	 * @param int $type
+	 *
+	 * @return string
+	 * @throws AppConfigIncorrectTypeException
+	 */
+	public function convertTypeToString(int $type): string {
+		$type &= ~self::VALUE_SENSITIVE;
+		try {
+			return match ($type) {
+				IAppConfig::VALUE_MIXED => 'mixed',
+				IAppConfig::VALUE_STRING => 'string',
+				IAppConfig::VALUE_INT => 'integer',
+				IAppConfig::VALUE_FLOAT => 'float',
+				IAppConfig::VALUE_BOOL => 'boolean',
+				IAppConfig::VALUE_ARRAY => 'array'
+			};
+		} catch (UnhandledMatchError) {
+			throw new AppConfigIncorrectTypeException('Unknown numeric type ' . $type);
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 *
+	 * @since 7.0.0
+	 */
+	public function deleteKey(string $app, string $key): void {
+		$this->assertParams($app, $key);
+		$qb = $this->connection->getQueryBuilder();
+		$qb->delete('appconfig')
+		   ->where($qb->expr()->eq('appid', $qb->createNamedParameter($app)))
+		   ->andWhere($qb->expr()->eq('configkey', $qb->createNamedParameter($key)));
+		$qb->executeStatement();
+
+		unset($this->lazyCache[$app][$key]);
+		unset($this->fastCache[$app][$key]);
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 *
+	 * @since 7.0.0
+	 */
+	public function deleteApp(string $app): void {
+		$this->assertParams($app);
+		$qb = $this->connection->getQueryBuilder();
+		$qb->delete('appconfig')
+		   ->where($qb->expr()->eq('appid', $qb->createNamedParameter($app)));
+		$qb->executeStatement();
+
+		$this->clearCache();
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param bool $reload set to TRUE to refill cache instantly after clearing it
+	 *
+	 * @since 29.0.0
+	 */
+	public function clearCache(bool $reload = false): void {
+		$this->lazyLoaded = $this->fastLoaded = false;
+		$this->lazyCache = $this->fastCache = $this->valueTypes = [];
+
+		if (!$reload) {
+			return;
+		}
+
+		$this->loadConfig(null);
+	}
+
+	/**
+	 * @inheritDoc
 	 * @return array
+	 * @since 29.0.0
 	 */
-	public function getFilteredValues($app) {
-		$values = $this->getValues($app, false);
+	public function statusCache(): array {
+		return [
+			'fastLoaded' => $this->fastLoaded,
+			'fastCache' => $this->fastCache,
+			'lazyLoaded' => $this->lazyLoaded,
+			'lazyCache' => $this->lazyCache,
+		];
+	}
 
-		if (isset($this->sensitiveValues[$app])) {
-			foreach ($this->sensitiveValues[$app] as $sensitiveKeyExp) {
-				$sensitiveKeys = preg_grep($sensitiveKeyExp, array_keys($values));
-				foreach ($sensitiveKeys as $sensitiveKey) {
-					$values[$sensitiveKey] = IConfig::SENSITIVE_VALUE;
-				}
+	/**
+	 * @param int $needle bitflag to search
+	 * @param int $type known value
+	 *
+	 * @return bool TRUE if bitflag $needle is set in $type
+	 */
+	private function isTyped(int $needle, int $type): bool {
+		return (($needle & $type) !== 0);
```

Comment:
```suggestion
		return (($needle & $type) === $needle);
```
This would support testing several flags at the same time I think

Reply:
no needed